tools/autoenv.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+# This script attempts to detect the correct environmental variables required
+# to build JSS from source.
+
+# Load OS information.
+if [ -f /etc/os-release ]; then
+    . /etc/os-release
+elif [ -f /usr/lib/os-release ];then
+    . /usr/lib/os-release
+else
+    echo "Cannot determine OS information. Exiting..." 1>&2
+    exit 1
+fi
+
+# Try and detect a JDK installation.
+if [ -d "/etc/alternatives/java_sdk" ]; then
+    export JAVA_HOME="/etc/alternatives/java_sdk"
+else
+    java8_jdk="$(find /usr/lib/jvm -maxdepth 1 -mindepth 1 -type d | grep '\(\-8\-\|-1\.8\.\)' | sort | tail -n 1)"
```

Comment:
Okay, so I changed this logic to try and detect `jdk8` automatically. This will probably be updated eventually with JDK-11 related stuff at which point we can do a more complete feature detection (`javac`, `java`, `jar`, etc. in the same bin directory), but it suffices for now on Fedora and Ubuntu.